src/Aspire.Hosting/Dcp/ApplicationExecutor.cs

```diff
@@ -637,14 +637,9 @@ ImmutableArray<int> GetPorts()
 
         ContainerLifetime GetContainerLifetime()
         {
-            if (container.AppModelResourceName is string resourceName &&
-                _applicationModel.TryGetValue(resourceName, out var appModelResource) &&
-                appModelResource.TryGetLastAnnotation<ContainerLifetimeAnnotation>(out var lifetimeAnnotation))
-            {
-                return lifetimeAnnotation.Lifetime;
-            }
-
-            return ContainerLifetime.Default;
+            return (container.Spec.Persistent ?? false)
+                ? ContainerLifetime.Persistent
+                : ContainerLifetime.Session;
```

Comment:
@drewnoakes FYI I noticed that this could be simplified to come from the container rather than the model.